src/implementations/title-escrow/changeHolder.test.ts

```diff
@@ -0,0 +1,104 @@
+import { TitleEscrowFactory, TradeTrustErc721Factory } from "@govtechsg/token-registry";
+import { Wallet } from "ethers";
+import { join } from "path";
+import { TitleEscrowChangeHolderCommand } from "../../commands/title-escrow/title-escrow-command.type";
+import { changeHolderOfTitleEscrow } from "./changeHolder";
+
+jest.mock("@govtechsg/token-registry");
+
+const changeHolderParams: TitleEscrowChangeHolderCommand = {
+  to: "0xabcd",
+  tokenId: "0xzyxw",
+  address: "0x1234",
+  network: "ropsten",
+  gasPriceScale: 1,
+  dryRun: false,
+};
+
+// TODO the following test is very fragile and might break on every interface change of TradeTrustErc721Factory
+// ideally must setup ganache, and run the function over it
+describe("title-escrow", () => {
+  describe("change holder of transferable record", () => {
+    const mockedTradeTrustErc721Factory: jest.Mock<TradeTrustErc721Factory> = TradeTrustErc721Factory as any;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore mock static method
+    const mockedConnectERC721: jest.Mock = mockedTradeTrustErc721Factory.connect;
+    const mockedTokenFactory: jest.Mock<TitleEscrowFactory> = TitleEscrowFactory as any;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore mock static method
+    const mockedConnectTokenFactory: jest.Mock = mockedTokenFactory.connect;
+    const mockedOwnerOf = jest.fn();
+    const mockChangeHolder = jest.fn();
+    const mockedTitleEscrowAddress = "0x2133";
+    mockedOwnerOf.mockReturnValue(mockedTitleEscrowAddress);
+    mockChangeHolder.mockReturnValue({
+      hash: "hash",
+      wait: () => Promise.resolve({ transactionHash: "transactionHash" }),
+    });
+
+    beforeEach(() => {
+      delete process.env.OA_PRIVATE_KEY;
+      mockedTradeTrustErc721Factory.mockReset();
+      mockedConnectERC721.mockReset();
+      mockedTokenFactory.mockReset();
+      mockedConnectTokenFactory.mockReset();
+      mockedConnectERC721.mockReturnValue({
```

Comment:
why is this in beforeEach while there is another block that setup mock return above ? (L33)

Reply:
hmm, yea i'll take a look at this, i mostly referenced the test structures from the other files.